WebRoot/js/zimbraAdmin/common/ZaTree.js

```diff
@@ -171,7 +180,11 @@ function(treeItem, skipNotify, kbNavEvent, noFocus) {
 
 	//this._expandUp(treeItem);
 	if (treeItem._setSelected(true, noFocus) && !skipNotify) {
-    	this._notifyListeners(DwtEvent.SELECTION, [treeItem], DwtTree.ITEM_SELECTED, null, this._selEv, kbNavEvent);
+        if (!isLicensedItem || isLicenseValid) {
+            this._notifyListeners(DwtEvent.SELECTION, [treeItem], DwtTree.ITEM_SELECTED, null, this._selEv, kbNavEvent);
```

Comment:
this event is not just related to storage management so blocking it based on license check is not valid

Reply:
Before the fix, `this._notifyListeners(...)` is executed on any item regardless of the condition of license check.
After the fix, it is executed if (a) the item is not related to any license (i.e. `!isLicensedItem`) or (b) it can be shown with a valid license (i.e. `isLicenseValid`)
The change does not break the current behavior on an item which is not related to any license as (a) is matched (unless `isLicensedItem = true;` at line 155 is executed) in the same way as https://github.com/Zimbra/zm-admin-console/pull/171 for `ZaTree.prototype._itemClicked`
Am I missing anything?

Reply:
Before the fix, `this._notifyListeners(...)` is executed on any item regardless of the condition of license check.
After the fix, it is executed if (a) the item is not related to any license (i.e. `!isLicensedItem`) or (b) it can be shown with a valid license (i.e. `isLicenseValid`)
The change does not break the current behavior on an item which is not related to any license as (a) is matched (unless `isLicensedItem = true;` at line 155 is executed) in the same way as https://github.com/Zimbra/zm-admin-console/pull/171 for `ZaTree.prototype._itemClicked`
Am I missing anything?

Reply:
I am still not able to understand this, so are we saying we will not fire SELECTION event for a tree item when license is not valid? what if we have a zimlet which wants to listen to this event without checking license status

Reply:
and can't the zimlet check license status instead of core application checking it?

Reply:
We have already defined the `ZaTree.licenseCheckArray = [];` blank array in this file. We are populating this array with objects through admin extensions that have license dependencies. In those specific cases, it will check the license, otherwise, the SELECTION event will work as before.
https://github.com/Zimbra/zm-admin-console/blob/ef1c5268a02ddbca9e6c80be9a0d096b942d7763/WebRoot/js/zimbraAdmin/common/ZaTree.js#L41

Reply:
@silentsakky sir, I haven't found a way to remove the for loop. When our app loads, the admin zimlets are called, and at that time, we push our licenseCheck objects into licenseArray. So, when we click on the tab and the itemClick method is triggered, it checks the array. According to my findings, we have to use a loop to iterate through that array. If you have any other suggestions or if I am missing something, please guide me on the same.